File: Aufgabe1/Insertion Sort/sort.py
x = [4,30,2,3,6,99,8,11,23]
def insertion_sort (x):
    for i in range(1, len(x)) :
        k = x[i]
        j = i - 1
        while(j >= 0 and x[j] > k):
            x[j+1] = x [j]
            j  -= 1
        x[j+1] = k
print(x)
insertion_sort (x)
print(x)

File: Aufgabe1/Insertion Sort/sort.rs

pub fn insertion_sort(x : &mut [i32]){
    let len = x.len();
    for i in 1..len {
        let k : i32 =  x[i];
        let mut j : usize =  i;
        while j > 0 && x[j - 1] > k {
            x[j] = x [j-1];
            j-=1;
        }
        x[j] = k;
    }
}